paywall/src/data-iframe/cache/localStorage.js

```diff
@@ -86,6 +124,58 @@ export async function put(window, networkId, accountAddress, type, value) {
   window.localStorage.setItem(key, JSON.stringify(container))
 }
 
+/**
+ * Retrieve the current cached account
+ *
+ * @param {object} window this is the global context, either global, window, or self
+ * @returns {string}
+ */
+export async function getAccount(window) {
+  if (!localStorageAvailable(window)) {
```

Comment:
It might be worthwhile to extract this into a method and us it in the methods calling the functionality below